arm-sql/2014-04-01/swagger/dataMasking.json

```diff
@@ -0,0 +1,460 @@
+                "suffixSize": {
+                    "type": "string",
+                    "description": "Gets or sets the suffixSize property that is used for the data masking rule."
+                },
+                "replacementString": {
+                    "type": "string",
+                    "description": "Gets or sets the replacementString property that is used for the data masking rule."
+                }
+            },
+            "required": [
+                "id",
+                "ruleState",
+                "maskingFunction"
+            ],
+            "description": "Represents the properties of an Azure SQL Database data masking rule."
+        },
+        "DataMaskingRule": {
+            "properties": {
+                "properties": {
+                    "x-ms-client-flatten": true,
+                    "$ref": "#/definitions/DataMaskingRuleProperties",
+                    "description": "Represents the properties of the resource."
+                }
+            },
+            "allOf": [
+                {
+                    "$ref": "#/definitions/SubResource"
+                }
+            ],
+            "description": "Represents an Azure SQL Database data masking rule."
+        },
+        "DataMaskingRuleListResult": {
+            "properties": {
+                "value": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/DataMaskingRule"
+                    },
+                    "description": "Gets or sets the list of Azure Sql Database data masking rules"
+                }
+            },
+            "description": "Represents the response to a List data masking rules request."
+        },
+        "Resource": {
+            "description": "Resource properties",
+            "properties": {
+                "name": {
+                    "readOnly": true,
+                    "type": "string",
+                    "description": "Resource name"
+                },
+                "id": {
+                    "readOnly": true,
+                    "type": "string",
+                    "description": "Resource Id"
+                },
+                "type": {
+                    "readOnly": true,
+                    "type": "string",
+                    "description": "Resource type"
+                },
+                "location": {
+                    "type": "string",
+                    "description": "Resource location"
+                },
+                "tags": {
+                    "type": "object",
+                    "additionalProperties": {
+                        "type": "string"
+                    },
+                    "description": "Resource tags"
+                }
+            },
+            "required": [
+                "location"
+            ],
+            "x-ms-azure-resource": true
+        },
+        "SubResource": {
+            "description": "Subresource properties",
```

Comment:
SubResource usually has an id property. I have not seen name as a part of the SubResource